### DISCUSSION TITLE
PurgeCSS support in Quasar v2

### DISCUSSION THREAD

**Discussion by bompus** [quasarframework/quasar#8692] [Ideas / Proposals] [answered]
Would it be possible to implement PurgeCSS into Quasar v2?

I noticed a related PR #6394 for v1 which seems to handle the text-blue-1 bg-yellow-2 etc concatenation issue mentioned on the previous discussion board but there doesn't seem to be any movement on the PR and I can't find any discussions or mention of PurgeCSS in Quasar v2 anywhere.

The latest version of PurgeCSS (4.x) requires PostCSS 8.x, but it appears that some other logic inside of Quasar needs to be updated to be compatible with PostCSS 8.x first as it errors when trying to force yarn resolutions to use postcss 8.

**Comment by marcelo-gondim** [comment]
By optimizing the codebase of the framework, we can reduce the amount of processing required to run Quasar-based applications, which in turn reduces energy consumption and contributes to a greener world. By prioritizing lean, efficient code in the development of the framework, we can help to ensure that Quasar remains a sustainable and environmentally responsible choice for developers.

We would like to request that the suggested improvement be added to the Quasar Framework roadmap, as we believe it would be a valuable addition to the framework's feature set. By including this optimization in a future release, we can further reduce the energy consumption of Quasar-based applications and help to build a more sustainable internet.

**Reply by 3m1n3nc3** [reply]
In support of a greener world.

**Reply by rstoenescu** [reply, marked as answer]
We are continuously optimizing and improving across the board, both CSS and JS -- runtime & footprint. We are all in for a greener world, however, PurgeCSS cannot be simply added out of the box and be expected to work as intended. In many cases, it requires additional configuration based on each use case -- and this latter part is why we can't add it by default. Unless keen on using it and assuming the extra time it requires for configuring it based on each change in your app code, the DX would decrease, sometimes significantly. Just like some of the developers don't want to use any linting (although we highly recommend it).

So bottom line, PurgeCSS can be manually added should anyone wish to do so. From a framework perspective, unless PurgeCSS can work reliably without the extra work, we won't add it out of the box.
